cli/lakeflags/flags.go

```diff
@@ -18,7 +20,7 @@ func (f *Flags) SetFlags(fs *flag.FlagSet) {
 }
 
 func (f *Flags) Names() (string, string) {
-	if i := strings.LastIndexByte(f.PoolName, '/'); i > -1 {
+	if i := strings.LastIndexByte(f.PoolName, '@'); i > -1 {
```

Comment:
I'd prefer to instead to have a `-ref` flag  instead of having to know you need to refer to a specific branch in the pool option written as `<pool>@<branch|commit>`. I think this might be a case where it's better to stick with how git works. Actually I'd be okay if you didn't have to specify the pool but if the `-ref` tag where written as `<pool>@<branch|commit>`, kind of like in git how you can write `refs/<remote>/<ref_name>`.

Reply:
See new discussion at end of `docs/lake/design.md`.  This is complicated enough that I vote we merge this PR and resolve in a subsequent and soonish followup PR...